packages/console/src/pages/TenantSettings/TenantBasicSettings/index.tsx

```diff
@@ -72,6 +78,34 @@ function TenantBasicSettings() {
     await saveData({ name, tag });
   });
 
+  const onClickDeletionButton = () => {
+    setIsDeletionModalOpen(true);
+  };
+
+  const onDelete = async () => {
+    if (isDeleting) {
+      return;
+    }
+
+    setIsDeleting(true);
+    try {
+      await cloudApi.delete(`/api/tenants/${currentTenantId}`);
+      setIsDeletionModalOpen(false);
+      void mutate();
+      if (tenants?.[0]?.id) {
+        window.open(new URL(`/${tenants[0].id}`, window.location.origin).toString(), '_self');
+      }
```

Comment:
What if it's the last tenant? Do we allow deleting tenant from the console if there's only one left now?

Reply:
Yes, will have a higher-level check on whether to direct the user to a page tailored for zero tenant scenario.